lane.py

```diff
@@ -194,10 +194,12 @@ def navigate(self, entrypoint):
         """Create a very similar FacetsWithEntryPoint that points to
         a different EntryPoint.
         """
-        return self.__class__(
+        destination = self.__class__(
             entrypoint=entrypoint, entrypoint_is_default=False,
             **self.constructor_kwargs
         )
+        destination.finalize_navigate(self)
```

Comment:
`navigate` is not a class method, so couldn't a subclass just override that without needing to add another method?

Reply:
You absolutely can, but I've always disliked how inheritance works for class methods. However, I might be using Python wrong. I'll take another look and see if I can improve this.

Reply:
But this isn't a class method anyway - only the other one is.

Reply:
It looks like I was able to take out both of these hook methods. Not the ones that decide which facets are available, though.